### ISSUE TITLE
Issue serializing IB orders

### ISSUE THREAD

**Issue by pcgm-team** [nautechsystems/nautilus_trader#1441]
I believe the bug can be recreated simply by sending any market orders on IB with streaming config defined.
```
2024-01-03T20:18:11.810040300Z [INF] TESTER-001.IntoViews-000: setting timer
2024-01-03T20:18:11.811402500Z [INF] TESTER-001.IntoViews-000: <--[EVT] OrderSubmitted(instrument_id=SCHW.NYSE, client_order_id=O-20240103-2018-001-000-2-E1, account_id=InteractiveBrokers-DU1435157, ts_event=1704313091810917901).
Could not convert '0.00' with type str: tried to convert to double
2024-01-03T20:18:11.874577200Z [WRN] TESTER-001.ExecEngine: Event message queue stopped with 1 message(s) on queue.
2024-01-03T20:18:16.911264400Z [WRN] TESTER-001.TradingNode: Timed out (5.0s) waiting for node to stop.
Status
------
DataEngine.check_disconnected() == False
ExecEngine.check_disconnected() == False
2024-01-03T20:18:16.911274800Z [INF] TESTER-001.TradingNode: DISPOSING...
2024-01-03T20:18:16.911440400Z [INF] TESTER-001.DataEngine: STOPPING...
2024-01-03T20:18:16.911449600Z [INF] TESTER-001.DataClient-InteractiveBrokers-001: STOPPING...
2024-01-03T20:18:16.912179100Z [INF] TESTER-001.DataClient-InteractiveBrokers-001: STOPPED.
2024-01-03T20:18:16.912589300Z [INF] TESTER-001.DataEngine: STOPPED.
2024-01-03T20:18:16.912596600Z [INF] TESTER-001.RiskEngine: STOPPING...
2024-01-03T20:18:16.912616000Z [INF] TESTER-001.RiskEngine: STOPPED.
2024-01-03T20:18:16.912619800Z [INF] TESTER-001.ExecEngine: STOPPING...
2024-01-03T20:18:16.912626300Z [INF] TESTER-001.ExecClient-InteractiveBrokers: STOPPING...
2024-01-03T20:18:16.912891500Z [INF] TESTER-001.ExecClient-InteractiveBrokers: STOPPED.
2024-01-03T20:18:16.913111600Z [INF] TESTER-001.ExecEngine: Canceling in-flight check task...
2024-01-03T20:18:16.913168600Z [INF] TESTER-001.ExecEngine: STOPPED.
2024-01-03T20:18:16.913178900Z [INF] TESTER-001.OrderEmulator: STOPPING...
2024-01-03T20:18:16.913191200Z [INF] TESTER-001.OrderEmulator: STOPPED.
2024-01-03T20:18:16.913198000Z [INF] TESTER-001.DataEngine: DISPOSING...
2024-01-03T20:18:16.913202900Z [INF] TESTER-001.DataClient-InteractiveBrokers-001: DISPOSING...
2024-01-03T20:18:16.913490700Z [INF] TESTER-001.DataClient-InteractiveBrokers-001: DISPOSED.
2024-01-03T20:18:16.913878800Z [INF] TESTER-001.DataEngine: DISPOSED.
2024-01-03T20:18:16.913891700Z [INF] TESTER-001.RiskEngine: DISPOSING...
2024-01-03T20:18:16.913896200Z [INF] TESTER-001.RiskEngine: DISPOSED.
2024-01-03T20:18:16.913900600Z [INF] TESTER-001.ExecEngine: DISPOSING...
2024-01-03T20:18:16.913903400Z [INF] TESTER-001.ExecClient-InteractiveBrokers: DISPOSING...
2024-01-03T20:18:16.914136000Z [INF] TESTER-001.ExecClient-InteractiveBrokers: DISPOSED.
2024-01-03T20:18:16.914346300Z [INF] TESTER-001.ExecEngine: DISPOSED.
2024-01-03T20:18:16.914366500Z [ERR] TESTER-001.TESTER-001: InvalidStateTrigger('RUNNING -> DISPOSE') state RUNNING.
2024-01-03T20:18:16.914369500Z [ERR] TESTER-001.TESTER-001: InvalidStateTrigger('RUNNING -> DISPOSE_COMPLETED') state RUNNING.
2024-01-03T20:18:16.916791800Z [INF] TESTER-001.TradingNode: Shutting down executor...
2024-01-03T20:18:16.916979700Z [INF] TESTER-001.TradingNode: Stopping event loop...
2024-01-03T20:18:16.917056800Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='Task-2' coro=<ActorExecutor._worker() running at C:\Users\USER\nautilus_trader\nautilus_trader\common\executor.py:165> wait_for=<Future pending cb=[Task.task_wakeup()]>>
2024-01-03T20:18:16.917079700Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='evt_queue' coro=<LiveRiskEngine._run_evt_queue() running at C:\Users\USER\nautilus_trader\nautilus_trader\live\risk_engine.py:267> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[gather.<locals>._done_callback() at C:\Users\USER\.conda\envs\for_live_strat\Lib\asyncio\tasks.py:754]>
INFO:ibapi.client:disconnecting
2024-01-03T20:18:16.917089200Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task cancelling name='inflight_check' coro=<LiveExecutionEngine._inflight_check_loop() running at C:\Users\USER\nautilus_trader\nautilus_trader\live\execution_engine.py:399> wait_for=<Future cancelled>>
2024-01-03T20:18:16.917099400Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='cmd_queue' coro=<LiveDataEngine._run_cmd_queue() running at C:\Users\CadeBruce\nautilus_trader\nautilus_trader\live\data_engine.py:382> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[gather.<locals>._done_callback() at C:\Users\USER\.conda\envs\for_live_strat\Lib\asyncio\tasks.py:754]>
2024-01-03T20:18:16.917129200Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='_run_incoming_msg_queue' coro=<InteractiveBrokersClient._run_incoming_msg_queue() running at C:\Users\USER\nautilus_trader\nautilus_trader\adapters\interactive_brokers\client\client.py:634> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[InteractiveBrokersClient._on_task_completed(None, None)()]>
2024-01-03T20:18:16.917151300Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='_run_incoming_msg_reader' coro=<InteractiveBrokersClient._run_incoming_msg_reader() running at C:\Users\USER\nautilus_trader\nautilus_trader\adapters\interactive_brokers\client\client.py:600> wait_for=<Future pending cb=[_chain_future.<locals>._call_check_cancel() at C:\Users\USER\.conda\envs\for_live_strat\Lib\asyncio\futures.py:387, Task.task_wakeup()]> cb=[InteractiveBrokersClient._on_task_completed(None, None)()]>
2024-01-03T20:18:16.917162800Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='data_queue' coro=<LiveDataEngine._run_data_queue() running at C:\Users\CadeBruce\nautilus_trader\nautilus_trader\live\data_engine.py:443> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[gather.<locals>._done_callback() at C:\Users\USER\.conda\envs\for_live_strat\Lib\asyncio\tasks.py:754]>
2024-01-03T20:18:16.917173000Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='Task-3' coro=<ActorExecutor._worker() running at C:\Users\USER\nautilus_trader\nautilus_trader\common\executor.py:165> wait_for=<Future pending cb=[Task.task_wakeup()]>>
2024-01-03T20:18:16.917190400Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='cmd_queue' coro=<LiveExecutionEngine._run_cmd_queue() running at C:\Users\USER\nautilus_trader\nautilus_trader\live\execution_engine.py:360> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[gather.<locals>._done_callback() at C:\Users\USER\.conda\envs\for_live_strat\Lib\asyncio\tasks.py:754]>
2024-01-03T20:18:16.917200000Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='res_queue' coro=<LiveDataEngine._run_res_queue() running at C:\Users\CadeBruce\nautilus_trader\nautilus_trader\live\data_engine.py:424> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[gather.<locals>._done_callback() at C:\Users\USER\.conda\envs\for_live_strat\Lib\asyncio\tasks.py:754]>
2024-01-03T20:18:16.917215300Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='_run_watch_dog' coro=<InteractiveBrokersClient._run_watch_dog() running at C:\Users\USER\nautilus_trader\nautilus_trader\adapters\interactive_brokers\client\client.py:435> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[InteractiveBrokersClient._on_task_completed(None, None)()]>
2024-01-03T20:18:16.917225500Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='cmd_queue' coro=<LiveRiskEngine._run_cmd_queue() running at C:\Users\CadeBruce\nautilus_trader\nautilus_trader\live\risk_engine.py:246> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[gather.<locals>._done_callback() at C:\Users\CadeBruce\.conda\envs\for_live_strat\Lib\asyncio\tasks.py:754]>
Traceback (most recent call last):
2024-01-03T20:18:16.917232300Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='Task-4' coro=<ActorExecutor._worker() running at C:\Users\CadeBruce\nautilus_trader\nautilus_trader\common\executor.py:165> wait_for=<Future pending cb=[Task.task_wakeup()]>>
2024-01-03T20:18:16.917240300Z [WRN] TESTER-001.TradingNode: Canceling pending task <Task pending name='req_queue' coro=<LiveDataEngine._run_req_queue() running at C:\Users\CadeBruce\nautilus_trader\nautilus_trader\live\data_engine.py:403> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[gather.<locals>._done_callback() at C:\Users\USER\.conda\envs\for_live_strat\Lib\asyncio\tasks.py:754]>
2024-01-03T20:18:16.917394100Z [WRN] TESTER-001.RiskEngine: Event message queue canceled.
2024-01-03T20:18:16.917396300Z [WRN] TESTER-001.RiskEngine: Event message queue stopped with 1 message(s) on queue.
    node.run()
2024-01-03T20:18:16.917403000Z [WRN] TESTER-001.DataEngine: DataCommand message queue canceled.
  File "C:\Users\USER\nautilus_trader\nautilus_trader\live\node.py", line 275, in run
2024-01-03T20:18:16.917404700Z [WRN] TESTER-001.DataEngine: DataCommand message queue stopped with 1 message(s) on queue.
2024-01-03T20:18:16.917767500Z [WRN] TESTER-001.DataEngine: Data message queue canceled.
    self.kernel.loop.run_until_complete(self.run_async())
2024-01-03T20:18:16.917770700Z [WRN] TESTER-001.DataEngine: Data message queue stopped with 1 message(s) on queue.
  File "C:\Users\USER\.conda\envs\for_live_strat\Lib\asyncio\base_events.py", line 653, in run_until_complete
2024-01-03T20:18:16.917784000Z [WRN] TESTER-001.ExecEngine: Command message queue canceled.
2024-01-03T20:18:16.917785400Z [WRN] TESTER-001.ExecEngine: Command message queue stopped with 1 message(s) on queue.
2024-01-03T20:18:16.917791400Z [WRN] TESTER-001.DataEngine: DataResponse message queue canceled.
2024-01-03T20:18:16.917792400Z [WRN] TESTER-001.DataEngine: DataResponse message queue stopped with 1 message(s) on queue.
    return future.result()
2024-01-03T20:18:16.917805700Z [WRN] TESTER-001.RiskEngine: Command message queue canceled.
           ^^^^^^^^^^^^^^^
2024-01-03T20:18:16.917807200Z [WRN] TESTER-001.RiskEngine: Command message queue stopped with 1 message(s) on queue.
  File "C:\Users\USER\nautilus_trader\nautilus_trader\live\node.py", line 319, in run_async
2024-01-03T20:18:16.917817600Z [WRN] TESTER-001.DataEngine: DataRequest message queue canceled.
2024-01-03T20:18:16.917818600Z [WRN] TESTER-001.DataEngine: DataRequest message queue stopped with 1 message(s) on queue.
2024-01-03T20:18:16.917875800Z [INF] TESTER-001.TWAP: TimeEvent(name=O-20240103-2018-001-000-2, event_id=324e7e1d-a160-4e6f-aba8-0fc64d85e6f9, ts_event=1704313094309918001, ts_init=1704313096917851501)
    await asyncio.gather(*tasks)
2024-01-03T20:18:16.918015400Z [WRN] TESTER-001.TradingNode: Can't find writer for cls: <class 'nautilus_trader.model.events.order.OrderInitialized'>
  File "C:\Users\USER\nautilus_trader\nautilus_trader\live\execution_engine.py", line 384, in _run_evt_queue
2024-01-03T20:18:16.918031900Z [INF] TESTER-001.IntoViews-000: <--[EVT] OrderInitialized(instrument_id=SCHW.NYSE, client_order_id=O-20240103-2018-001-000-2-E2, side=BUY, type=MARKET, quantity=13, time_in_force=GTC, post_only=False, reduce_only=False, quote_quantity=False, options={}, emulation_trigger=NO_TRIGGER, trigger_instrument_id=None, contingency_type=NO_CONTINGENCY, order_list_id=None, linked_order_ids=None, parent_order_id=None, exec_algorithm_id=TWAP, exec_algorithm_params=None, exec_spawn_id=O-20240103-2018-001-000-2, tags=None).
    self._handle_event(event)
2024-01-03T20:18:16.918113000Z [INF] TESTER-001.TWAP: [CMD]--> SubmitOrder(order=MarketOrder(BUY 13 SCHW.NYSE MARKET GTC, status=INITIALIZED, client_order_id=O-20240103-2018-001-000-2-E2, venue_order_id=None, position_id=None, exec_algorithm_id=TWAP, exec_spawn_id=O-20240103-2018-001-000-2, tags=None), position_id=None).
  File "nautilus_trader\\execution\\engine.pyx", line 848, in nautilus_trader.execution.engine.ExecutionEngine._handle_event
2024-01-03T20:18:16.918363300Z [INF] TESTER-001.TradingNode: Closing event loop...
2024-01-03T20:18:16.918500200Z [INF] TESTER-001.TradingNode: loop.is_running=False
2024-01-03T20:18:16.918501600Z [INF] TESTER-001.TradingNode: loop.is_closed=True
2024-01-03T20:18:16.918502000Z [INF] TESTER-001.TradingNode: DISPOSED.
  File "nautilus_trader\\execution\\engine.pyx", line 904, in nautilus_trader.execution.engine.ExecutionEngine._handle_event
  File "nautilus_trader\\execution\\engine.pyx", line 1011, in nautilus_trader.execution.engine.ExecutionEngine._apply_event_to_order
  File "nautilus_trader\\execution\\engine.pyx", line 1011, in nautilus_trader.execution.engine.ExecutionEngine._apply_event_to_order
  File "nautilus_trader\\execution\\engine.pyx", line 1024, in nautilus_trader.execution.engine.ExecutionEngine._apply_event_to_order
  File "nautilus_trader\\common\\component.pyx", line 1199, in nautilus_trader.common.component.MessageBus.publish_c
  File "C:\Users\USER\nautilus_trader\nautilus_trader\persistence\writer.py", line 246, in write
    serialized = ArrowSerializer.serialize_batch([obj], data_cls=cls)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\nautilus_trader\nautilus_trader\serialization\arrow\serializer.py", line 171, in serialize_batch
    batches = [ArrowSerializer.serialize(obj, data_cls) for obj in data]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\nautilus_trader\nautilus_trader\serialization\arrow\serializer.py", line 171, in <listcomp>
    batches = [ArrowSerializer.serialize(obj, data_cls) for obj in data]
... (4 lines left)

```

**Comment by pcgm-team** [comment]
Thank you, look forward to this fix.

**Comment by pcgm-team** [comment]
Update:
The only relevant root error is actually:
```Could not convert '0.00' with type str: tried to convert to double```.

**Comment by pcgm-team** [comment]
If it is helpful, the debug log is:

```
 2024-01-24T15:20:46.040018271Z [DEBUG] TESTER-001.InteractiveBrokersClient-001: Msg handled: function=nextValidId data={'order_id': 179}
 2024-01-24T15:20:46.068841087Z [DEBUG] TESTER-001.InteractiveBrokersClient-001: Msg buffer received: b'\x00\x00\x00+3\x00179\x00Submitted\x000\x00824\x000\x00193966283\x000\x000\x001\x00\x000\x00'
 2024-01-24T15:20:46.068867984Z [DEBUG] TESTER-001.InteractiveBrokersClient-001: Msg buffer received: b''
 2024-01-24T15:20:46.068953319Z [DEBUG] TESTER-001.InteractiveBrokersClient-001: Msg received: b'5\x00179\x0010098\x00BAC\x00STK\x00\x000\x00?\x00\x00SMART\x00USD\x00BAC\x00BAC\x00BUY\x00824\x00MKT\x000.0\x000.0\x00GTC\x00\x00DU1435157\x00\x000\x00O-20240124-1520-001-000-1:179\x001\x00193966283\x000\x000\x000\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x000\x00\x00-1\x000\x00\x00\x00\x00\x00\x002147483647\x000\x000\x000\x00\x003\x000\x000\x00\x000\x000\x00\x000\x00None\x00\x000\x00\x00\x00\x00?\x000\x000\x00\x000\x000\x00\x00\x00\x00\x00\x000\x000\x000\x002147483647\x002147483647\x00\x00\x000\x00\x00IB\x000\x000\x00\x000\x000\x00Submitted\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x00\x00\x00\x00\x00\x000\x000\x000\x00None\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x001.7976931348623157E308\x000\x00\x00\x00\x000\x001\x000\x000\x000\x00\x00\x000\x00\x00100\x000.02\x00\x00\x00'
2024-01-24T15:20:46.068972335Z [DEBUG] TESTER-001.InteractiveBrokersClient-001: Msg received fields: (b'5', b'179', b'10098', b'BAC', b'STK', b'', b'0', b'?', b'', b'SMART', b'USD', b'BAC', b'BAC', b'BUY', b'824', b'MKT', b'0.0', b'0.0', b'GTC', b'', b'DU1435157', b'', b'0', b'O-20240124-1520-001-000-1:179', b'1', b'193966283', b'0', b'0', b'0', b'', b'', b'', b'', b'', b'', b'', b'', b'', b'', b'', b'0', b'', b'-1', b'0', b'', b'', b'', b'', b'', b'2147483647', b'0', b'0', b'0', b'', b'3', b'0', b'0', b'', b'0', b'0', b'', b'0', b'None', b'', b'0', b'', b'', b'', b'?', b'0', b'0', b'', b'0', b'0', b'', b'', b'', b'', b'', b'0', b'0', b'0', b'2147483647', b'2147483647', b'', b'', b'0', b'', b'IB', b'0', b'0', b'', b'0', b'0', b'Submitted', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'', b'', b'', b'', b'', b'0', b'0', b'0', b'None', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'1.7976931348623157E308', b'0', b'', b'', b'', b'0', b'1', b'0', b'0', b'0', b'', b'', b'0', b'', b'100', b'0.02', b'', b'')
2024-01-24T15:20:46.069240683Z [DEBUG] TESTER-001.InteractiveBrokersClient-001: Msg handled: function=openOrder data={'order_id': 179, 'contract': 139900004386320: 10098,BAC,STK,,0,?,,SMART,,USD,BAC,BAC,False,,,,combo:, 'order': 139900152942608: 179,1,193966283: MKT BUY 824@0 GTC, 'order_state': <ibapi.order_state.OrderState object at 0x7f3d0aeaeb10>}
2024-01-24T15:20:46.069387604Z [DEBUG] TESTER-001.InteractiveBrokersClient-001: Msg received: b'3\x00179\x00Submitted\x000\x00824\x000\x00193966283\x000\x000\x001\x00\x000\x00'
2024-01-24T15:20:46.069402229Z [DEBUG] TESTER-001.InteractiveBrokersClient-001: Msg received fields: (b'3', b'179', b'Submitted', b'0', b'824', b'0', b'193966283', b'0', b'0', b'1', b'', b'0')
2024-01-24T15:20:46.069422238Z [DEBUG] TESTER-001.InteractiveBrokersClient-001: Msg handled: function=orderStatus data={'order_id': 179, 'status': 'Submitted', 'filled': Decimal('0'), 'remaining': Decimal('824'), 'avg_fill_price': 0.0, 'perm_id': 193966283, 'parent_id': 0, 'last_fill_price': 0.0, 'client_id': 1, 'why_held': '', 'mkt_cap_price': 0.0}
 2024-01-24T15:20:46.069428170Z [DEBUG] TESTER-001.ExecClient-INTERACTIVE_BROKERS: Ignoring `_on_order_status` event for order_status='Submitted' is handled in `_on_open_order`
Could not convert '0.00' with type str: tried to convert to double
```

**Comment by pcgm-team** [comment]
You can see that the 'mkt_cap_price': 0.0, last_fill_price': 0 and 'avg_fill_price': 0.0, which may have something to do with it.

**Comment by benjaminsingleton** [comment]
I believe I've found the source of the issue. In a dummy strategy that I created, an `OrderUpdated` event caused the same error. I noticed that `trigger_price=Price('0.000')` seemed to be related to issue and generated the same `Could not convert ... with type str: tried to convert to double` error.

The issue is that the arrow schema defined for `OrderUpdated` has `"trigger_price": pa.float64()`, which is different from `"price": pa.string()`. So I changed the schema type for `"trigger_price": pa.string()` and I didn't have the issue any more.

https://github.com/nautechsystems/nautilus_trader/blob/master/nautilus_trader/serialization/arrow/schema.py#L346

It's also worth noting that `trigger_price` is defined as a string here: https://github.com/nautechsystems/nautilus_trader/blob/master/nautilus_trader/serialization/arrow/schema.py#L146

I'll submit a PR.
